### ISSUE TITLE
RobotFramework folder location not as documented & AutoItX3.dll not in documented location

### ISSUE THREAD

**Issue by medic8601** [nokia/robotframework-autoitlibrary#33]
Windows 10 Pro 10.0.18363 Build 18363 64 bit Virtual Machine
Python 3.9.6

I install robotframework-autoitlibrary from requirements file.

There are two issues:

1.  The RobotFramework folder is being created in the site-packages folder NOT in C:\
2. The AutoItX3.dll folder is not being placed in the site-packages/AutoItLibrary/lib folder.. There is NOT a lib folder at all

This is causing the following error on import of the library:

```
C:\Users\SQAAdmin>robot -d *** -v CHROMEDRIVER:94 -v BRANCH:Main -v ***:*** -v ***:*** -t *** ***\RemoteMachineSuite.robot
[ ERROR ] Error in file '***\resources.robot' on line 2: Importing library 'AutoItLibrary' failed: com_error: (-2147312566, 'Error loading type library/DLL.', None, None)
Traceback (most recent call last):
  File "c:\python39\lib\site-packages\AutoItLibrary\__init__.py", line 46, in <module>
    win32com.client.gencache.EnsureModule("{F8937E53-D444-4E71-9275-35B64210CC3B}", 0, 1, 0)
  File "c:\python39\lib\site-packages\win32com\client\gencache.py", line 515, in EnsureModule
    module = MakeModuleForTypelib(typelibCLSID, lcid, major, minor, progressInstance, bForDemand = bForDemand, bBuildHidden = bBuildHidden)
  File "c:\python39\lib\site-packages\win32com\client\gencache.py", line 284, in MakeModuleForTypelib
    makepy.GenerateFromTypeLibSpec( (typelibCLSID, lcid, major, minor), progressInstance=progressInstance, bForDemand = bForDemand, bBuildHidden = bBuildHidden)
  File "c:\python39\lib\site-packages\win32com\client\makepy.py", line 223, in GenerateFromTypeLibSpec
    tlb = pythoncom.LoadRegTypeLib(typelibCLSID, major, minor, lcid)
PYTHONPATH:
  C:\Python39\Scripts\robot.exe
  C:\***\Framework
  C:\***\Framework\lib
  C:\***\AutomatedTests
  C:\Users\***
  c:\python39\python39.zip
  c:\python39\DLLs
  c:\python39\lib
  c:\python39
  c:\python39\lib\site-packages
  c:\python39\lib\site-packages\win32
  c:\python39\lib\site-packages\win32\lib
  c:\python39\lib\site-packages\Pythonwin
```

Attempts (all done by admin user in admin prompt):

1. I have fully purged all packages and reinstalled from requirements.txt file - no help
2. I have uninstalled robotframework-autoitlibrary and reinstalled manually on cli - no help
        a. Noted that the pip uninstall fails with this error
```
Uninstalling robotframework-autoitlibrary-1.2.7:
ERROR: Exception:
Traceback (most recent call last):
  File "c:\python39\lib\shutil.py", line 806, in move
    os.rename(src, real_dst)
PermissionError: [WinError 5] Access is denied: 'c:\\python39\\lib\\site-packages\\robotframework\\extensions\\autoitlibrary\\' -> 'C:\\Python39\\Lib\\site-packages\\RobotFramework\\Extensions\\~utoitlibrary'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python39\Lib\site-packages\pip\_internal\cli\base_command.py", line 164, in exc_logging_wrapper
    status = run_func(*args)
  File "C:\Python39\Lib\site-packages\pip\_internal\commands\uninstall.py", line 97, in run
    uninstall_pathset = req.uninstall(
  File "C:\Python39\Lib\site-packages\pip\_internal\req\req_install.py", line 696, in uninstall
    uninstalled_pathset.remove(auto_confirm, verbose)
  File "C:\Python39\Lib\site-packages\pip\_internal\req\req_uninstall.py", line 384, in remove
    moved.stash(path)
  File "C:\Python39\Lib\site-packages\pip\_internal\req\req_uninstall.py", line 282, in stash
    renames(path, new_path)
  File "C:\Python39\Lib\site-packages\pip\_internal\utils\misc.py", line 307, in renames
    shutil.move(old, new)
  File "c:\python39\lib\shutil.py", line 824, in move
    rmtree(src)
  File "c:\python39\lib\shutil.py", line 740, in rmtree
    return _rmtree_unsafe(path, onerror)
  File "c:\python39\lib\shutil.py", line 622, in _rmtree_unsafe
    onerror(os.rmdir, path, sys.exc_info())
  File "c:\python39\lib\shutil.py", line 620, in _rmtree_unsafe
    os.rmdir(path)
OSError: [WinError 145] The directory is not empty: 'c:\\python39\\lib\\site-packages\\robotframework\\extensions\\autoitlibrary\\'
```
3. After pip install, i have manually created the lib folder and copied the dll there and the ran the registration command and the win32 makepy command found in setup.py - no help
4. After another uninstall, I have cloned the repo and installed using `python setup.py install` - This works as documented. The RobotFramework folder is created in the C:\ drive and the lib folder is created in the site-packages/AutoItLibrary folder and is populated with the AutoItX3.dll file. A subsequent execution of our test runs without error.

There is apparently a severe issue with the pip install process.

**Comment by medic8601** [comment]
After further investigation it appear to be and issue with when pip installs the package using the .whl file and the wheel package. Below is the console output from testing. In it I purge the cache and pip install this package (wheel is not installed) so it uses the legacy setup.py install method. After this install all files/folders are in the correct locations as per documentation. 

Then i uninstall this package, install wheel, purge the cache, and pip install this package. After this install, the RobotFramework folder is back in the site-packages folders and the lib folder is not created in the site-packages/AutoItLibrary folder.

```
C:\Users\SQAadmin>pip cache purge
Files removed: 209

C:\Users\SQAadmin>pip install robotframework-autoitlibrary
Collecting robotframework-autoitlibrary
  Downloading robotframework-autoitlibrary-1.2.7.tar.gz (696 kB)
     |████████████████████████████████| 696 kB 3.3 MB/s
Requirement already satisfied: pywin32 in c:\python39\lib\site-packages (from robotframework-autoitlibrary) (300)
Collecting pillow
  Downloading Pillow-8.3.2-cp39-cp39-win_amd64.whl (3.2 MB)
     |████████████████████████████████| 3.2 MB ...
Using legacy 'setup.py install' for robotframework-autoitlibrary, since package 'wheel' is not installed.
Installing collected packages: pillow, robotframework-autoitlibrary
    Running setup.py install for robotframework-autoitlibrary ... done
Successfully installed pillow-8.3.2 robotframework-autoitlibrary-1.2.7
WARNING: You are using pip version 21.1.3; however, version 21.3 is available.
You should consider upgrading via the 'c:\python39\python.exe -m pip install --upgrade pip' command.

C:\Users\SQAadmin>pip uninstall -y robotframework-autoitlibrary
Found existing installation: robotframework-autoitlibrary 1.2.7
Uninstalling robotframework-autoitlibrary-1.2.7:
  Successfully uninstalled robotframework-autoitlibrary-1.2.7

C:\Users\SQAadmin>pip install wheel
Collecting wheel
  Downloading wheel-0.37.0-py2.py3-none-any.whl (35 kB)
Installing collected packages: wheel
Successfully installed wheel-0.37.0
WARNING: You are using pip version 21.1.3; however, version 21.3 is available.
You should consider upgrading via the 'c:\python39\python.exe -m pip install --upgrade pip' command.

C:\Users\SQAadmin>pip cache purge
Files removed: 7

C:\Users\SQAadmin>pip install robotframework-autoitlibrary
Collecting robotframework-autoitlibrary
  Downloading robotframework-autoitlibrary-1.2.7.tar.gz (696 kB)
     |████████████████████████████████| 696 kB 3.3 MB/s
Requirement already satisfied: pywin32 in c:\python39\lib\site-packages (from robotframework-autoitlibrary) (300)
Requirement already satisfied: pillow in c:\python39\lib\site-packages (from robotframework-autoitlibrary) (8.3.2)
Building wheels for collected packages: robotframework-autoitlibrary
  Building wheel for robotframework-autoitlibrary (setup.py) ... done
  Created wheel for robotframework-autoitlibrary: filename=robotframework_autoitlibrary-1.2.7-py3-none-any.whl size=485632 sha256=29a99953cb031ca5592eb3326c64f331f2f4681d1f61d7a97b7f12d2f1f36b7d
  Stored in directory: c:\users\sqaadmin\appdata\local\pip\cache\wheels\1e\5c\ff\d8d976a79817f66fb16cfa585c34768bc1c803aab4ba8fdfb4
Successfully built robotframework-autoitlibrary
Installing collected packages: robotframework-autoitlibrary
Successfully installed robotframework-autoitlibrary-1.2.7
WARNING: You are using pip version 21.1.3; however, version 21.3 is available.
You should consider upgrading via the 'c:\python39\python.exe -m pip install --upgrade pip' command.
```

I'll also note that in the above tests when uninstalling using the legacy method it failed to delete the RobotFramework and AutoItLibrary folders from the filesystem

**Comment by medic8601** [comment]
Relates to issue #28 
Any traction on this?

**Comment by lucyking** [comment]
> Relates to issue #28 Any traction on this?

@medic8601 fix by v1.2.9
